openapi-cli/src/main/java/org/ballerinalang/openapi/utils/TypeExtractorUtil.java

```diff
@@ -163,24 +164,21 @@ public static BallerinaOpenApiType extractOpenApiObject(OpenAPI apiDef, Filter f
      * @param operation     ballerinaOpenApiOperation object
      * @throws BallerinaOpenApiException throws ballerina openApi exception
      */
-    private static void setFilteredOperation(String pathName, Map.Entry<PathItem.HttpMethod, Operation> nextOp,
-                                             Operation opObject, BallerinaOpenApiOperation operation)
+    private static void setFilteredOperation(BallerinaOpenApiPath pathName, Map.Entry<PathItem.HttpMethod,
+            Operation> nextOp, Operation opObject, BallerinaOpenApiOperation operation)
             throws BallerinaOpenApiException {
 
-        if (opObject.getOperationId() == null) {
-            String resName = "resource_" + nextOp.getKey().toString().toLowerCase(Locale.ENGLISH)
-                    + pathName.replaceAll("/", "_")
-                    .replaceAll("[{}]", "");
-            operation.setOpName(resName);
-            outStream.println("warning : `" + resName + "` is used as the resource name since the " +
-                    "operation id is missing for " + pathName + " " + nextOp.getKey());
-        } else {
-            operation.setOpName(escapeIdentifier(
-                    opObject.getOperationId().replace(" ", "_")));
-        }
-
+        operation.setOpName(pathName.getPath().replaceFirst("/", ""));
         if (opObject.getParameters() != null) {
-            operation.setParameterList(extractOpenApiParameters(opObject.getParameters()));
+            List<BallerinaOpenApiParameter> parameters = extractOpenApiParameters(opObject.getParameters());
+            for (BallerinaOpenApiParameter parameter : parameters) {
+                if (parameter.isPathParam()) {
+                    operation.setOpName(pathName.getPath().replace("{" + parameter.getParamName() + "}",
```

Comment:
Shall create variables for these 2 args and pass the variables as this looks a bit cluttered.